lib/galaxy/model/orm/engine_factory.py

```diff
@@ -43,17 +43,25 @@ def pretty_stack():
     return rval
 
 
-def build_engine(url, engine_options, database_query_profiling_proxy=False, trace_logger=None, slow_query_log_threshold=0, thread_local_log=None, log_query_counts=False):
+def build_engine(
+    url,
+    engine_options,
+    database_query_profiling_proxy=False,
+    trace_logger=None,
+    slow_query_log_threshold=0,
+    thread_local_log=None,
+    log_query_counts=False
+):
     if database_query_profiling_proxy or slow_query_log_threshold or thread_local_log or log_query_counts:
 
-        @event.listens_for(Engine, "before_execute")
-        def before_execute(conn, clauseelement, multiparams, params):
+        @event.listens_for(Engine, "before_cursor_execute")
+        def before_cursor_execute(conn, cursor, statement, parameters, context, executemany):
             conn.info.setdefault('query_start_time', []).append(time.time())
 
     if slow_query_log_threshold or thread_local_log or log_query_counts:
+
         @event.listens_for(Engine, "after_cursor_execute")
-        def after_cursor_execute(conn, cursor, statement,
-                                 parameters, context, executemany):
+        def after_cursor_execute(conn, cursor, statement, parameters, context, executemany):
```

Comment:
I'd move the non-functional stylistic changes into their own PR ahead of this one but that isn't a big deal if the final PR turns out to be this focused and managable.

Reply:
I'll do that. (This one change above though only makes sense together with the main edit: I made it so the two event handlers are visually consistent (after/before_cursor_execute) now that they have the same signature.)

Reply:
I've removed the non-essential stylistic change (build_engine definition)